File: trunk/PolarMath/src/org/macondo/polar/evaluation/statistics/RMSSD.java
package org.macondo.polar.evaluation.statistics;

import java.util.List;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;

/**
 * <p></p>
 *
 * Date: 19.05.2008
 *
 * @author <a href="mailto:ktsibriy@gmail.com">Kirill Y. Tsibriy</a>
 */
public class RMSSD implements Evaluation<Integer> {
    public Integer evaluate(Training training) {
        final List<Integer> intervals = training.getIntervals();
        long total = 0;

        for (int i = 1; i < intervals.size(); i++) {
            Integer now = intervals.get(i);
            Integer before = intervals.get(i-1);

            total += (now - before) * (now - before);
        }

        return (int) Math.sqrt(total / intervals.size());
    }
}

File: trunk/PolarMathCS/PolarMath/Evaluation/Statistics/RMSSD.cs
using PolarMath.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PolarMath.Evaluation.Statistics
{
    public class RMSSD : IEvaluation<int>
    {
        public int Evaluate(Training training) 
        {
            var intervals = training.Intervals;
            long total = 0;

            for (var i = 1; i < intervals.Count; i++)
            {
                var now = intervals.ElementAt( i );
                var before = intervals.ElementAt( i - 1 );

                total += (now - before) * (now - before);
            }

            return (int) Math.Sqrt( total / (double)intervals.Count );
        }
    }
}
